File: Frontend-inmocasas/app/src/main/java/com/example/inmocasas/dto/Inmueble.kt
package com.example.inmocasas.dto

class Inmueble (
    var id: Int = 0,
    var createdAt: Long = 0,
    var updatedAt: Long = 0,
    var sector: String = "",
    var callePrincipal: String ="",
    var calleSecundaria: String = "",
    var numeroPisos: Int = 0,
    var numeroHabitaciones:Int = 0,
    var numeroBanios:Int = 0,
    var numeroParqueaderos:Int = 0,
    var antiguedad: String = "",
    var ciudad: String = "",
    var fkPublicacion: Int = 0
){

}

File: backend-inmocasas/backend-inmocasas/api/models/Inmueble.js
/**
 * Inmueble.js
 *
 * @description :: A model definition represents a database table/collection.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {
    sector:{
      type:'string',
      required:true

    },
    callePrincipal:{
      type:'string',
    },
    calleSecundaria:{
      type:'string',
    },
    numeroPisos:{
      type:'number',
    },
    numeroHabitaciones:{
      type:'number',
    },
    numeroBaños:{
      type:'number',
    },
    numeroParqueaderos:{
      type:'number',
    },
    antiguedad:{
      type:'string',
    },
    ciudad:{
      type: 'string',
    },
    fkPublicacion:{
      model:'publicacion',
      unique:true
    }
  },
};

